scripts/fires/wfigs_data.py

```diff
@@ -0,0 +1,278 @@
+                                         id=row.CpxID[1:-1].lower())
+        else:
+            return None
+
+    df["FireDiscoveryDateTime"] = df["FireDiscoveryDateTime"].apply(
+        get_datetime)
+    df["Year"] = df["FireDiscoveryDateTime"].apply(lambda x: x.year)
+    df["FireDiscoveryDateTime"] = df["FireDiscoveryDateTime"].apply(
+        get_date_str)
+    df = df[df["Year"] >= 2014]
+    df["ContainmentDateTime"] = df["ContainmentDateTime"].apply(
+        get_datetime).apply(get_date_str)
+    df["ControlDateTime"] = df["ControlDateTime"].apply(get_datetime).apply(
+        get_date_str)
+    df["InitialResponseDateTime"] = df["InitialResponseDateTime"].apply(
+        get_datetime).apply(get_date_str)
+    df["Costs"] = df["EstimatedCostToDate"].apply(get_cost)
+    df['BurnedArea'] = df['DailyAcres'].apply(get_area,
+                                              args=('daily_acres',
+                                                    _DAILY_ACRES_MAP))
+    df["FireCause"] = df["FireCause"].apply(get_cause)
+    df["FireCauseGeneral"] = df["FireCauseGeneral"].apply(get_cause)
+    df["FireCauseSpecific"] = df["FireCauseSpecific"].apply(get_cause)
+    df["TotalIncidentPersonnel"] = df["TotalIncidentPersonnel"].apply(
+        get_personnel)
+    df["ParentFire"] = df.apply(get_parent_fire, axis=1)
+    df['InitialResponseAcres'] = df['InitialResponseAcres'].apply(
+        get_area, args=('initial_response_area', _INITIAL_RESPONSE_AREA_MAP))
+
+    df["Location"] = df.apply(get_place, axis=1)
+    df["FIPS"] = df.apply(get_fips, axis=1)
+    df["Location"] = df["Location"].fillna(df["FIPS"])
+    df.loc[df["Location"] == '', 'Location'] = df["FIPS"]
+    df["IrwinID"] = df["IrwinID"].apply(lambda x: x.lower())
+    df["dcid"] = df["IrwinID"].apply(
+        lambda x: "{prefix}{id}".format(prefix=_FIRE_DCID_PREFIX, id=x))
+    df["name"] = df["IncidentName"].apply(lambda x: re.sub(
+        ' +', ' ',
+        x.replace("\n", "").replace("'", "").replace('"', "").replace('[', ""))
+                                          + ' Fire').apply(lambda x: x.title())
+    df["typeOf"] = df["IncidentTypeCategory"].apply(
+        lambda x: _FIRE_INCIDENT_MAP[x])
+    df["wfigsFireID"] = df["UniqueFireIdentifier"]
+    col_list = [
+        "dcid", "name", "typeOf", "Location", "FireCause", "FireCauseGeneral",
+        "FireCauseSpecific", "FireDiscoveryDateTime", "ControlDateTime",
+        "ContainmentDateTime", "BurnedArea", "Costs", "TotalIncidentPersonnel",
+        "IrwinID", "wfigsFireID", "ParentFire", "InitialResponseDateTime",
+        "InitialResponseAcres"
+    ]
+    return df[col_list]
+
+
+def main(_) -> None:
+    pre_2022_df = get_data(PRE_2022_FIRE_LOCATIONS_URL)
+    df = pre_2022_df
+    df = process_df(df)
+    df.to_csv("final_processed_data.csv", index=False)
+    with open('location_file.json', 'w') as locations:
+        locations.write(json.dumps(_CACHE))
```

Comment:
Please hide reading/writing this cache behind a flag. We also should not check that file in.

Reply:
Done